### PR TITLE
feat(docs): add telemetry page

### LINKED ISSUE #2089: Add telemetry landing page
Take the logic from the [telemetry package](https://github.ibm.com/carbon/ibm/blob/main/packages/telemetry/src/telemetry.js), and create a landing page that we can link to from the CLI and/or logs.

References:

* https://www.gatsbyjs.com/docs/telemetry
* https://nextjs.org/telemetry

### PR DESCRIPTION
**Pull request by mattrosno**
Closes #2089

Adds a telemetry landing page to support the `@carbon/telemetry` package that we're rolling out to the Carbon packages.

### CODE REVIEW

**Comment by mattrosno** [comment]
@joshblack can you review to ensure this page is factually correct? Also, I kept the "why are we collecting telemetry data" pretty minimal. Let me know if you think we need to elaborate more, to the likes of https://www.gatsbyjs.com/docs/telemetry and https://nextjs.org/telemetry.

@janchild this is a page that we link to from CLI and logs when our packages are installed. Let me know if you think it should live in a different location besides the "Help" section.

**Review by joshblack — APPROVED** [review]

File: src/pages/help/telemetry.mdx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+### Why is telemetry collected?
+
+Carbon collects telemetry information to influence Carbon's roadmap and
+prioritize bug fixes.
+
+### When is data collected?
+
+Data is only collected when Carbon packages are installed on a Continuous
+Integration (CI) server, when the installation is not part of a Pull Request
+(PR), for Git respositories with the origin remotes:
```
Comment (janchild) [inline src/pages/help/telemetry.mdx:22]:
```suggestion
(PR), and for Git repositories with the origin remotes:
```

Comment (janchild) [reply src/pages/help/telemetry.mdx:22]: @mattrosno It looks like a word was missing here. Could you pls check to make sure I've kept your intended meaning?

File: src/pages/help/telemetry.mdx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+- `github.ibm.com`
+- `github.com/ibm`
+- `github.com/carbon-design-system`
+
+As the information collected is not anonymous, telemetry data collection is
+limited to IBM GitHub Enterprise repositories, as well as open-source
```
Comment (janchild) [inline src/pages/help/telemetry.mdx:29]:
```suggestion
limited to IBM GitHub Enterprise repositories, as well as open source
```

**Review by janchild — COMMENTED** [review]
@mattrosno  Looks good and I think this position works. Just a couple of tweaks.

**Comment by janchild** [comment]
@mattrosno Thinking through further where this information should live... The FAQs would be a good place for this content and it would save adding a new page to the IA. We are using H2s on this page, so from a formatting point of view it works, and the content is already set up in Qs and As.

**Comment by mattrosno** [comment]
@janchild ok I'll move to FAQ, and make sure we link to this URL in telemetry logging instead.

https://www.carbondesignsystem.com/help/faq#telemetry

File: src/pages/help/telemetry.mdx (hunk 1 above)
Comment (mattrosno) [reply src/pages/help/telemetry.mdx:22]: Thank you.

**Comment by janchild** [comment]
@mattrosno Looks good! Just need to remove the .yaml file. :)

**Review by janchild — APPROVED** [review]

**Pull request merged**